banzai/images.py

```diff
@@ -88,32 +105,26 @@ def writeto(self, filename, fpack=False):
         image_hdu.header['EXTEND'] = True
         image_hdu.name = 'SCI'
         hdu_list = [image_hdu]
-        if self.catalog is not None:
-            table_hdu = fits_utils.table_to_fits(self.catalog)
-            table_hdu.name = 'CAT'
-            hdu_list.append(table_hdu)
-        if self.bpm is not None:
-            bpm_hdu = fits.ImageHDU(self.bpm.astype(np.uint8))
-            bpm_hdu.name = 'BPM'
-            hdu_list.append(bpm_hdu)
+        hdu_list = self.add_bpm_to_hdu_list(hdu_list)
+        hdu_list = self.add_data_tables_to_hdu_list(hdu_list)
 
-        hdu_list = fits.HDUList(hdu_list)
+        fits_hdu_list = fits.HDUList(hdu_list)
```

Comment:
As Curtis mentioned, maybe this needs to be refactored a bit (and also I'm not sure he wanted a separate BPM adding method). Perhaps lines 107 to 123 could be moved to their own method. And then you can just go back to calling this variable `hdu_list`.

Reply:
the catalog business is necessary to get banzai-nres working so it can move forward again. I would like to leave refactoring of the write_to function after this HDUList thing for another PR. What do you think? For now, I will change the variable back to hdu_list.